src-browserify/api/sql.js

```diff
@@ -1,686 +1,697 @@
 var _ = require('underscore');
-var _Promise = require('_Promise');
 var Mustache = require('mustache');
 
-// TODO: jQuery.ajax OR reqwest is expected to be located at global namespace when an SQL object is created
-function SQL(options) {
-  if(cartodb === this || window === this) {
-    return new SQL(options);
-  }
-  if(!options.user) {
-    throw new Error("user should be provided");
-  }
-  var loc = new String(window.location.protocol);
-  loc = loc.slice(0, loc.length - 1);
-  if(loc == 'file') {
-    loc = 'https';
-  }
-
-  this.ajax = options.ajax || (typeof(jQuery) !== 'undefined' ? jQuery.ajax: reqwest);
-  if(!this.ajax) {
-    throw new Error("jQuery or reqwest should be loaded");
-  }
-
-  this.options = _.defaults(options, {
-    version: 'v2',
-    protocol: loc,
-    jsonp: typeof(jQuery) !== 'undefined' ? !jQuery.support.cors: false
-  })
-
-  if (!this.options.sql_api_template) {
-    var opts = this.options;
-    var template = null;
-    if(opts && opts.completeDomain) {
-      template = opts.completeDomain;
-    } else {
-      var host = opts.host || 'cartodb.com';
-      var protocol = opts.protocol || 'https';
-      template = protocol + '://{user}.' + host;
+// NOTE this does not return a SQL directly, but a wrapper, to inject the dependencies
+// e.g. var SQL = require('./sql')({ jQuery: $ /* or reqwest: reqwest */ });
+// @param {Object} _Promise
+// @param {Object} opts hash with keys:
+//   jQuery: {Object} (Optional) has precedence over reqwest if both are present
+//   reqwest: {Object} (Optional)
+module.exports = function(_Promise, opts) {
```

Comment:
Read the PR description? I tried to explain the approach there. Anyway, I'll try to clarify further with this particular case since asked:
`_Promise` itself relies on `Backbone.Events` _or_ `BackboneIsch.Events` (the fake Backbone object used in core to not load the full Backbone lib, see `api/core-lib/backbone-isch.js`). Which one is used? Depends on the context/bundle (core or standard), so to avoid having to set this up in every place used and send around unnecessary dependencies (e.g. SQL don't use Backbone) this is wired up in the [core entry file](https://github.com/CartoDB/cartodb.js/pull/727/files#diff-36f4cca05a9602edf8347e27fedf45e5R33) (the standard bundle will be setup differently).